src/entries/popup/pages/home/TokenDetails/PriceChart.tsx

```diff
@@ -148,6 +148,35 @@ const fetchPriceChart = async (
     }, [] as ChartData[]) ?? null
   );
 };
+type PriceChartQueryKeyArgs = {
+  address: AddressOrEth;
+  chainId: ChainId;
+  time?: ChartTime;
+};
+const priceChartQueryKey = ({
+  address,
+  chainId,
+  time,
+}: PriceChartQueryKeyArgs) => ['price chart', { address, chainId, time }];
+export function getPriceChartQueryCache({
+  address,
+  chainId,
+  time,
+}: PriceChartQueryKeyArgs) {
+  return queryClient.getQueriesData({
```

Comment:
if you pass a type parameter here like `getQueriesData<QueryKeyType>()` you won't need to do the type assertion on line 169

Reply:
the generic types the data, not the key

Reply:
no 😔